### PR TITLE
Separate strategies' requirements

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

moving Horovod and the additional strategy requirement to own file
updating and cleaning all dockers
update CI workflows
**Fixes:** failing HPU testing

### Does your PR introduce any breaking changes? If yes, please list them.

Horovod is not installed with `.[extra]` but with `.[strategies]`

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda @tchaton @rohitgr7

### CODE REVIEW

**Comment by Borda** [comment]
have an n00b question, would be possible to unify the package name and Cuda-version have in a version like PT, for example, `torch==1.11.0+cu113`  instead of `bagua-cuda113==0.9.0`? this makes it impossible to include in PL requirements directly...
cc: @wangraying @kaushikb11

File: .azure-pipelines/hpu-tests.yml
Comment (Borda) [inline .azure-pipelines/hpu-tests.yml:36]: this shall be the correct fix for #12726
cc: @akihironitta

Comment (akihironitta) [reply .azure-pipelines/hpu-tests.yml:36]: Thank you for the suggestion! Will update the PR!

**Comment by akihironitta** [comment]
I like the idea of not putting everything inside `extra.txt`!

Can be done in a separate PR, but I'm also considering if we should split the requirements into smaller pieces so that people can install dependencies of their interest and nothing more. Do you think it's reasonable to have the following installation options for example?

```shell
pip install pytorch-lightning[fsdp]
pip install pytorch-lightning[bagua]
pip install pytorch-lightning[horovod]
...

pip install pytorch-lightning[strategies]  # to install all of the above
```

**Comment by Borda** [comment]
> Can be done in a separate PR, but I'm also considering if we should split the requirements into smaller pieces so that people can install dependencies of their interest and nothing more. Do you think it's reasonable to have the following installation options for example?

that would be considering as next step :)

File: .github/workflows/ci_test-full.yml
Comment (akihironitta) [inline .github/workflows/ci_test-full.yml:118]: nit: this is simpler?
```suggestion
          grep "horovod" requirements/strategies.txt > requirements/horovod.txt
```

File: requirements/test.txt
Comment (akihironitta) [inline requirements/test.txt:10]: This entry seems a duplication of a below line `scikit-learn>0.22.1`. Can be removed in this PR?
```suggestion
```

**Review by akihironitta — APPROVED** [review]
LGTM!

File: requirements/test.txt
Comment (carmocca) [inline requirements/test.txt:11]: great!

File: .azure-pipelines/hpu-tests.yml
Comment (carmocca) [inline .azure-pipelines/hpu-tests.yml:38]: nit: prefer TODO since it's what I usually grep for

```suggestion
        pip uninstall -y horovod  # TODO: not sure how it got there...
```

**Review by carmocca — APPROVED** [review]
Great improvement

**Comment by Borda** [comment]
```
/usr/local/lib/python3.7/dist-packages/_pytest/runner.py:340: in from_call
    result: Optional[TResult] = func()
/usr/local/lib/python3.7/dist-packages/_pytest/runner.py:371: in <lambda>
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
/usr/local/lib/python3.7/dist-packages/_pytest/doctest.py:545: in collect
    module = import_path(self.path, root=self.config.rootpath)
/usr/local/lib/python3.7/dist-packages/_pytest/pathlib.py:553: in import_path
    raise ImportPathMismatchError(module_name, module_file, path)
```
not sure about the source :/

**Review by awaelchli — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

File: tests/standalone_tests.sh
Comment (carmocca) [inline tests/standalone_tests.sh:38]: This is not a TODO. It's done on purpose because it needs to be run separately.


```suggestion
# tests to skip - space separated
```

File: tests/standalone_tests.sh
Comment (carmocca) [inline tests/standalone_tests.sh:39]: Why are we ignoring this?

Comment (Borda) [reply tests/standalone_tests.sh:39]: Trying to find why the stand alone does not find tests...

Comment (carmocca) [reply tests/standalone_tests.sh:39]: Oh yeah, just realized we call it manually below

**Comment by carmocca** [comment]
There was an error with GPU CI installing deepspeed:

```python
Collecting deepspeed>=0.6.0
  Downloading deepspeed-0.6.2.tar.gz (542 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 542.4/542.4 KB 21.5 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Traceback (most recent call last):
        File "<string>", line 36, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-3rgf_22_/deepspeed_cc308bd058f44eaa93222e60c059a30f/setup.py", line 35, in <module>
          from op_builder import ALL_OPS, get_default_compute_capabilities, OpBuilder
      ImportError: cannot import name 'OpBuilder' from 'op_builder' (/usr/local/lib/python3.7/dist-packages/op_builder/__init__.py)
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

```

**Comment by akihironitta** [comment]
`deepspeed==0.6.2` has some trouble with its installation: https://github.com/microsoft/DeepSpeed/issues/1918

File: requirements/strategies.txt
Comment (akihironitta) [inline requirements/strategies.txt:2]: Regarding the deepspeed version `deepspeed>=0.6.0`,

- isn't this too strict? It should support `<0.6.0` as well, no? cc @SeanNaren
- 0.6.0 and 0.6.1 are incompatible due to https://github.com/microsoft/DeepSpeed/issues/1878.
- 0.6.2 is incompatible due to its installation issue https://github.com/microsoft/DeepSpeed/issues/1918.

File: .azure-pipelines/gpu-tests.yml
Comment (akihironitta) [inline .azure-pipelines/gpu-tests.yml:56]: this deepspeed reinstallation can be removed as https://github.com/microsoft/DeepSpeed/issues/1878 is now resolved in the latest release 0.6.3.
```suggestion
```

Comment (akihironitta) [reply .azure-pipelines/gpu-tests.yml:56]: Tried 0.6.3 in this CI run https://dev.azure.com/PytorchLightning/pytorch-lightning/_build/results?buildId=69283&view=logs&j=3afc50db-e620-5b81-6016-870a6976ad29, and there seems no problem.

Comment (Borda) [reply .azure-pipelines/gpu-tests.yml]: setting `deepspeed>=0.6.3`

**Review by Borda — COMMENTED** [review]
go

File: tests/standalone_tests.sh
Comment (carmocca) [inline tests/standalone_tests.sh:27]: Remove debug statement

```suggestion
```

File: .azure-pipelines/gpu-tests.yml
Comment (Borda) [reply .azure-pipelines/gpu-tests.yml:54]:
```suggestion
```

**Review by Borda — COMMENTED** [review]
.

**Review by otaj — APPROVED** [review]

File: requirements/strategies.txt
Comment (akihironitta) [inline requirements/strategies.txt:2]:
```suggestion
deepspeed!=0.6.0,!=0.6.1,!=0.6.2
```

Comment (akihironitta) [reply requirements/strategies.txt:2]: or
```suggestion
deepspeed<0.6.0,>=0.6.3
```
(not sure if this is valid though.)

Comment (Borda) [reply requirements/strategies.txt:2]: let's keep this simple as this is the actual state on master and do this adjustments in separate PR :rabbit:

**Comment by akihironitta** [comment]
@williamFalcon Could you have a look as codeowner?

**Review by Borda — COMMENTED** [review]
..

**Comment by akihironitta** [comment]
@Borda To merge this PR, we need @williamFalcon's approval or admin power, right?

**Comment by Borda** [comment]
> @Borda To merge this PR, we need @williamFalcon's approval or admin power, right?

or @lexierule

File: requirements/strategies.txt
Comment (kaushikb11) [reply requirements/strategies.txt:2]: @akihironitta @Borda Let's create a Github issue for this so that we don't miss it?

**Review by Borda — COMMENTED** [review]
goo

**Pull request merged**
